x-pack/plugins/ml/public/application/services/new_job_capabilities/new_job_capabilities_service_analytics.ts

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { Field, NewJobCapsResponse } from '../../../../common/types/fields';
+import { ES_FIELD_TYPES, IIndexPattern } from '../../../../../../../src/plugins/data/public';
+import { processTextAndKeywordFields, NewJobCapabilitiesServiceBase } from './new_job_capabilities';
+import { ml } from '../ml_api_service';
+
+// Keep top nested field and remove all <nested_field>.* fields
+function removeNestedFieldChildren(resp: NewJobCapsResponse, indexPatternTitle: string) {
+  const results = resp[indexPatternTitle];
+  const fields: Field[] = [];
+  const nestedFields: Record<string, boolean> = {};
+
+  if (results !== undefined) {
+    results.fields.forEach((field: Field) => {
+      if (field.type === ES_FIELD_TYPES.NESTED && nestedFields[field.name] === undefined) {
+        nestedFields[field.name] = true;
+        fields.push(field);
+      }
+    });
+
+    if (Object.keys(nestedFields).length > 0) {
+      results.fields.forEach((field: Field) => {
+        if (field.type !== ES_FIELD_TYPES.NESTED) {
+          const fieldNameParts = field.name.split('.');
+          const rootOfField = fieldNameParts.shift();
+          if (rootOfField && nestedFields[rootOfField] === undefined) {
+            fields.push(field);
+          }
+        }
+      });
+    } else {
+      fields.push(...results.fields);
+    }
+  }
+  return fields;
+}
+
+class NewJobCapsServiceAnalytics extends NewJobCapabilitiesServiceBase {
+  constructor() {
+    super();
+  }
+
+  public async initializeFromIndexPattern(indexPattern: IIndexPattern) {
+    try {
+      const resp: NewJobCapsResponse = await ml.dataFrameAnalytics.analyticsFields(
+        indexPattern.title,
+        indexPattern.type === 'rollup'
+      );
+
+      const allFields = removeNestedFieldChildren(resp, indexPattern.title);
+
+      const { fieldsPreferringKeyword } = processTextAndKeywordFields(allFields);
+
+      // set the main fields list to contain fields which have been filtered to prefer
+      // keyword fields over text fields.
+      // e.g. if foo.keyword and foo exist, don't add foo to the list.
+      this._fields = fieldsPreferringKeyword;
```

Comment:
although they're not used at the moment, I think it still would be good to create the aggregations too. 
This would mean replacing the `aggIds` lists with the real aggs and cross reference them with the fields. 
This way the all `NewJobCapabilitiesServiceBase` derived classes produce the same response, a fully cross referenced list of fields and aggs.